sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -1352,7 +1352,9 @@ def _create_and_write_workflow(self,
                                                          pipeline_loop_crs, recursive_tasks_names)
         inlined_as_taskSpec.extend(e)
         workflow['spec']['pipelineSpec']['tasks'] = workflow_tasks
-
+        # Preserve order of params, required by tests.
+        if 'params' in workflow['spec']:
+          workflow['spec']['params'] = sorted(workflow['spec']['params'], key=lambda kv: (kv['name']))
```

Comment:
So we already have this logic but just for step 1 of inline task. This is for after step 2.

Reply:
Thanks @pugangxa , That one sorts inside a CR `pipeline_loop_crs[i]['spec']['pipelineSpec']['params']` and this one sorts `PipelineRun.Spec.Params`.